### PR TITLE
Add protobuf-c: C library for protobuf manipulation

### PR DESCRIPTION
**Pull request by hobu**
Checklist
- [x] Title of this PR is meaningful: e.g. "Adding my_nifty_package", not "updated meta.yaml".
- [x] License file is packaged (see [here](https://github.com/conda-forge/staged-recipes/blob/5eddbd7fc9d1502169089da06c3688d9759be978/recipes/example/meta.yaml#L64-L73) for an example).
- [x] Source is from official source.
- [x] Package does not vendor other packages. (If a package uses the source of another package, they should be separate packages or the licenses of all packages need to be packaged).
- [ ] If static libraries are linked in, the license of the static library is packaged.
- [x] Package does not ship static libraries. If static libraries are needed, [follow CFEP-18](https://github.com/conda-forge/cfep/blob/main/cfep-18.md).
- [x] Build number is 0.
- [x] A tarball (`url`) rather than a repo (e.g. `git_url`) is used in your recipe (see [here](https://conda-forge.org/docs/maintainer/adding_pkgs.html#build-from-tarballs-not-repos) for more details).
- [x] GitHub users listed in the maintainer section have posted a comment confirming they are willing to be listed there.
- [ ] When in trouble, please check our [knowledge base documentation](https://conda-forge.org/docs/maintainer/knowledge_base.html) before pinging a team.

### CODE REVIEW

**Comment by hobu** [comment]
This PR will need https://github.com/conda-forge/abseil-cpp-feedstock/pull/80 to be merged to successfully build the Windows package.

File: recipes/protobufc/bld.bat [hunk 1]
```diff
@@ -0,0 +1,16 @@
...
+cd build-cmake
+mkdir build
+cd build
+
+cmake -G "Ninja" ^
+      %CMAKE_ARGS% ^
+      -DCMAKE_INSTALL_PREFIX:PATH="%LIBRARY_PREFIX%" ^
+      -DCMAKE_BUILD_TYPE:STRING=Release ^
+      -DCMAKE_LIBRARY_PATH="%LIBRARY_LIB%" ^
+      -DCMAKE_INCLUDE_PATH="%INCLUDE_INC%" ^
```
Comment (isuruf) [inline recipes/protobufc/bld.bat:11]:
```suggestion
```

File: recipes/protobufc/meta.yaml [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+    - test -f ${PREFIX}/lib/libprotobuf-c${SHLIB_EXT}
+
+about:
+  home: https://github.com/protobuf-c/protobuf-c
+  summary: 'C API for Google Protobuf'
+  description: |
+    a pure C library that implements protobuf encoding and decoding
+  license: BSD-3-Clause
+  license_family: BSD
+  license_file: LICENSE.txt
```
Comment (isuruf) [inline recipes/protobufc/meta.yaml:39]:
```suggestion
  license_file: LICENSE
```

File: recipes/protobufc/LICENSE.txt [hunk 3]
```diff
@@ -0,0 +1,32 @@
+Copyright (c) 2008-2023, Dave Benson and the protobuf-c authors.
```
Comment (isuruf) [inline recipes/protobufc/LICENSE.txt:1]: This file is not needed

File: recipes/protobufc/bld.bat (hunk 1 above)
Comment (hobu) [reply recipes/protobufc/bld.bat]: updated as suggested in 53920f3b508d2968e2db5c0c1e80f8bd41575a76

File: recipes/protobufc/LICENSE.txt (hunk 3 above)
Comment (hobu) [reply recipes/protobufc/LICENSE.txt]: updated as suggested in c48273d8a0c4fb9955ed0ffe76b612b7906d724b

File: recipes/protobufc/meta.yaml [hunk 4]
```diff
@@ -0,0 +1,42 @@
...
+
+package:
+  name: {{ name|lower }}
+  version: {{ version }}
+
+source:
+  url: https://github.com/protobuf-c/protobuf-c/archive/refs/tags/v{{ version }}.tar.gz
+  sha256: d4cb022d55f49796959b07a9d83040822e39129bc0eb28f4e8301da17d758f62
+  patches:
+    - patches/0001-squash-clean-cmake.patch
```
Comment (isuruf) [inline recipes/protobufc/meta.yaml:12]: Where's this patch coming from?

File: recipes/protobufc/meta.yaml [hunk 5]
```diff
@@ -0,0 +1,44 @@
+{% set name = "protobufc" %}
```
Comment (h-vetinari) [inline recipes/protobufc/meta.yaml:1]: can we name the package `protobuf-c`? It's not the name of the binary, but it matches the upstream project name, and it distinguishes more easily from `protobuf`.

File: recipes/protobufc/meta.yaml [hunk 6]
```diff
@@ -0,0 +1,44 @@
...
+    - {{ stdlib("c") }}
+    - cmake
+    - ninja
+  host:
+    - libprotobuf
+    - libabseil
+
+test:
+  commands:
+    - test -f ${PREFIX}/lib/libprotobuf-c${SHLIB_EXT}
```
Comment (h-vetinari) [inline recipes/protobufc/meta.yaml:31]: If we're building for windows, you'll need to split this. Also, there's a binary `protoc-gen-c` being packaged that we should test.
```suggestion
    - test -f ${PREFIX}/lib/libprotobuf-c${SHLIB_EXT}   # [unix]
    - if not exist %LIBRARY_BIN%/protobuf-c.dll         # [win]
    - protoc-gen-c --version
```

File: recipes/protobufc/meta.yaml (hunk 4 above)
Comment (hobu) [reply recipes/protobufc/meta.yaml]: comment added pointing to it.

File: recipes/protobufc/meta.yaml (hunk 5 above)
Comment (hobu) [reply recipes/protobufc/meta.yaml]: renamed as suggested

File: recipes/protobufc/meta.yaml (hunk 6 above)
Comment (hobu) [reply recipes/protobufc/meta.yaml]: added as suggested.

**Comment by hobu** [comment]
@conda-forge/help-c-cpp this is ready for review.

**Pull request merged**